### PR TITLE
Fix 13096 - Imported private identifiers conflict with public ones

### PR DESCRIPTION
**Pull request by yglukhov**
Fix [13096](https://issues.dlang.org/show_bug.cgi?id=13096)

### CODE REVIEW

**Comment by quickfur** [comment]
LGTM

**Comment by AndrejMitrovic** [comment]
Pinging @9rnsr again since he knows more about this.

**Comment by MetaLang** [comment]
The fact that this bug has been around since 2007 is crazy. This needs merging post-haste.

**Comment by 9rnsr** [comment]
Please add the test case that I've shown.

File: src/dsymbol.c [hunk 1]
```diff
@@ -978,6 +978,11 @@ Dsymbol *ScopeDsymbol::search(Loc loc, Identifier *ident, int flags)
                             a = mergeOverloadSet(a, s2);
                             continue;
                         }
+                        else if (ss->isModule() && s2->prot() == PROTprivate)
+                        {
+                            continue;
+                        }
```
Comment (9rnsr) [inline src/dsymbol.c]: This is not bad, but does cover all cases?

1. `ss` is an overloadable public symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

2. `ss` is an overloadable private symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

3. `ss` is an non-overloadable public symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

4. `ss` is an non-overloadable priate symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

**Comment by mathias-lang-sociomantic** [comment]
This issue have been fixed. I think we can safely close this.

**Pull request closed without merging**